components/ntp_background_images/browser/view_counter_service.cc

```diff
@@ -96,6 +111,9 @@ ViewCounterService::GetCurrentBrandedWallpaperData() const {
 }
 
 base::Value ViewCounterService::GetCurrentWallpaperForDisplay() const {
+  for (auto& observer : observer_list_)
```

Comment:
This is definitely not the right place to put it. Sometimes, it worked as expected. Sometimes, it was getting multiple calls per page load

Reply:
Right, this method can be called anytime.
Curious the difference between `OnBrandedWallpaperShown` and `OnWallpaperShown`?
What we want from `OnWallpapaerShown`?

Reply:
Actually- I named it badly 😛 I just pushed an update which you can check now - I renamed as `OnNewTabOpened`. This is what we really need to find out 😄